src/pipeline.js

```diff
@@ -4,54 +4,54 @@ export const makeImporter = (importHooks, moduleCache = new Map()) => {
     resolve,
     locate,
     retrieve,
-    rewrite,
+    analyze,
     rootLinker,
   } = importHooks;
 
-  function recursiveLink(moduleId, linker, preEndowments) {
+  function recursiveLink(moduleLocation, linker, preEndowments) {
     const subLinker =
-      (linker.linkerFor && linker.linkerFor(moduleId)) || linker;
+      (linker.linkerFor && linker.linkerFor(moduleLocation)) || linker;
 
-    let moduleInstance = subLinker.instanceCache.get(moduleId);
+    let moduleInstance = subLinker.instanceCache.get(moduleLocation);
     if (moduleInstance) {
       return moduleInstance;
     }
 
     // Translate the linkage record into a module instance, and cache it.
-    const linkageRecord = moduleCache.get(moduleId);
+    const linkageRecord = moduleCache.get(moduleLocation);
```

Comment:
I was confused by this line because the term "cache" normally means that it might always miss, and we should always be prepared for it to miss. None of our `*Cache`s are caches. `*Memo`s? In any case, don't rename in this PR.